Automattic/wp-calypso
# Themes - Fix "back" to Themes showcase on empty query.

#### Changes proposed in this Pull Request

Added showcase boolean to data store:
* Showcase stays open when going "back" with no query params.
* Showcase will automatically be set open on data store if loaded with query params, which allows returning to the themes showcase with no query to already be open even if they had never seen the button due to the original load w/ query.

#### Testing instructions



* Run this PR locally.
* Navigate to themes section.
* Open "More Themes" and do NOT enter any query params.
* Scroll down and select "Info" for some theme.
* Hit the "back" button.
* Verify we are back at the theme we selected with the showcase open.

* STILL TODO - Hit "Info" again, and hit the "Back" button again.  Verify this no longer leads us back to the theme we selected and that we are now facing the top of the page (Recommended Themes).

Fixes #38120 (partial fix currently)
